libs/docker-client/src/main/kotlin/batect/docker/pull/RegistryCredentialsSource.kt

```diff
@@ -45,10 +45,10 @@ data class BasicCredentialsSource(val encodedCredentials: String, val serverAddr
                 throw DockerRegistryCredentialsException("Decoded credentials for '$serverAddress' are not valid.")
             }
 
-            if (parts[0] == "<token>") {
-                return TokenRegistryCredentials(parts[1], serverAddress)
+            return if (parts[0] == "<token>") {
```

Comment:
This might be even clearer as a `when`.

Reply:
Personally, I avoid using `when` if there only one condition and an `else`. To me, that's the typical `if` use-case, and `when`'s power lies in using a subject, which usually does not make sense in this case.